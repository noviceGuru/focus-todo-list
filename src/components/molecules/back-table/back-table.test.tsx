import { cleanup, render, screen } from "@testing-library/react"
import { test, expect, afterEach } from "vitest"

import BackTable from "components/molecules/back-table/back-table"

const startingTodos = [
    {
        id: '1',
        task: 'Task number one'
    },
    {
        id: '2',
        task: 'Task number two'
    },
]

afterEach(() => cleanup())

test("The table renders with no toods, it shows the All Done note", () => {
    render(<BackTable todos={[]} setTodos={() => {}} />)
    const allDoneNote = screen.getByText(/all done!/i)

    expect(allDoneNote).toBeTruthy()
})

test('The table shows the todos that it\'s been given, and matches the snapshot', ()=>{
    render(<BackTable todos={startingTodos} setTodos={() => {}} />)

    const row1 = screen.getByText(/task number one/i)
    const row2 = screen.getByText(/task number two/i)
    const table = screen.getByTestId('table')

    expect(row1).toBeTruthy()
    expect(row2).toBeTruthy()
    expect(table).toMatchSnapshot()
})
