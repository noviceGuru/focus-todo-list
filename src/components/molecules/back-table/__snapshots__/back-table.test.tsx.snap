// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The table shows the todos that it's been given, and matches the snapshot 1`] = `
<div
  class="rounded-xl overflow-hidden table-auto"
  data-testid="table"
>
  <div>
    <div
      class=" border-y-2 h-12"
    >
      <div
        class="flex h-full items-center justify-between p-1 bg-lime-300 cursor-pointer hover:bg-slate-300 line-clamp-1 transition-colors"
      >
        <span
          class="truncate pl-1"
          title="Task number one"
        >
          Task number one
        </span>
        <button
          class="bg-pink-300 p-1 rounded-full opacity-0 hover:opacity-100"
        >
          <img
            class="w-8"
            src="/focus-todo-list/src/assets/clear-one.svg"
            title="Delete Task"
          />
        </button>
      </div>
    </div>
    <div
      class=" border-y-2 h-12"
    >
      <div
        class="flex h-full items-center justify-between p-1 bg-lime-300 cursor-pointer hover:bg-slate-300 line-clamp-1 transition-colors"
      >
        <span
          class="truncate pl-1"
          title="Task number two"
        >
          Task number two
        </span>
        <button
          class="bg-pink-300 p-1 rounded-full opacity-0 hover:opacity-100"
        >
          <img
            class="w-8"
            src="/focus-todo-list/src/assets/clear-one.svg"
            title="Delete Task"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
